frontend/src/app/core/helpers/time-helper.ts

```diff
@@ -1,6 +1,35 @@
 import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+import { DayAction } from '@core/enums/day-action.enum';
 
-export function TimeRangeValidator(firstControl: AbstractControl): ValidatorFn {
-    return (control: AbstractControl): ValidationErrors | null =>
-        (firstControl?.value < control?.value ? null : { error: true });
+export const TimeRangeValidator =
+    (firstControl: AbstractControl): ValidatorFn =>
+        (control: AbstractControl): ValidationErrors | null =>
+            (firstControl?.value < control?.value ? null : { error: true });
+
+export function normalizeHours(hours: number): [number, DayAction] {
+    if (hours < 0) {
+        return [24 + hours, DayAction.SubtractDay];
+    }
+    if (hours >= 24) {
+        return [hours - 24, DayAction.AddDay];
+    }
+
+    return [hours, DayAction.NoAction];
+}
```

Comment:
Why do we need this?

Reply:
I convert string hours to user selected time zone. After adding timeZone hours to time range value, hours can become negative or more than 24. This used to give hours appropriate values

Reply:
Fixed